stan/math/rev/core/operator_addition.hpp

```diff
@@ -99,7 +121,7 @@ inline var operator+(var a, Arith b) {
   if (b == 0.0) {
     return a;
   }
-  return {new internal::add_vd_vari(a.vi_, b)};
+  return {new internal::add_vari<decltype(a.vi_), double>(a.vi_, b)};
```

Comment:
Won't these types be inferred?

Reply:
What do you mean by inferred? For classes I think you always need to declare the template types. In C++17 you can just call the constructor and it will infer the type